--- drivers/usb/usb.c
+++ drivers/usb/usb.c
@@ -312,7 +312,7 @@
         vxair_mac_write32(slot_id, 0x0088, val32 & ~(1 << 0)); // Clear B_AX_PLATFORM_EN
         vxair_mac_write32(slot_id, 0x0088, val32 | (1 << 0));  // Set B_AX_PLATFORM_EN
 
-        tx_desc->mac_ax.qsel = 0;
+        // --- mac_enable_cpu (dlfw = 1) ---
         vxair_mac_write32(slot_id, 0x01E8, 0); // R_AX_LDM (0x01E8)
         vxair_mac_write32(slot_id, 0x0160, 0); // R_AX_HALT_H2C_CTRL (0x0160)
         vxair_mac_write32(slot_id, 0x0164, 0); // R_AX_HALT_C2H_CTRL (0x0164)
@@ -384,6 +384,7 @@
         tx_desc->mac_ax.macid = 0x80;
         tx_desc->mac_ax.pckt_type = 1;
         tx_desc->mac_ax.ls = 1;
+        tx_desc->mac_ax.qsel = 0;
         tx_desc->mac_ax.qpn = 0;
         tx_desc->mac_ax.amsdu = 0;
         tx_desc->mac_ax.wlan_hdr_len = 0;
